core/domain/suggestion_services.py

```diff
@@ -0,0 +1,367 @@
+
+    Args:
+        suggestion_type: str. The type of the suggestion.
+        target_type: str. The target entity being edited.
+
+        (The above 2 parameters Should be one of the constants defined in
+        storage/suggestion/gae_models.py.)
+
+        target_id: str. The ID of the target entity being suggested to.
+        target_version_at_submission: int. The version number of the target
+            entity at the time of creation of the suggestion.
+        author_id: str. The ID of the user who submitted the suggestion.
+        change_cmd: dict. The actual content of the suggestion.
+        score_category: str. The category to score the suggestor on.
+        description: str. The description of the changes provided by the author.
+        assigned_reviewer_id: str(optional). The ID of the user assigned to
+            review the suggestion.
+        final_reviewer_id: str(optional). The ID of the reviewer who has
+        accepted/rejected the suggestion.
+    """
+    if target_type == suggestion_models.TARGET_TYPE_EXPLORATION:
+        thread_id = feedback_services.create_thread(
+            target_id, None, author_id, description,
+            DEFAULT_SUGGESTION_THREAD_SUBJECT)
+
+        # This line will be removed after the feedback thread migration is
+        # complete.
+        thread_id = suggestion_models.TARGET_TYPE_EXPLORATION + '.' + thread_id
+    suggestion_models.SuggestionModel.create(
+        suggestion_type, target_type, target_id, target_version_at_submission,
+        suggestion_models.STATUS_IN_REVIEW, author_id, assigned_reviewer_id,
+        final_reviewer_id, change_cmd, score_category, thread_id)
+
+
+def get_suggestion_from_model(suggestion_model):
+    """Converts the given SuggestionModel to a Suggestion object
+
+    Args:
+        suggestion_model: SuggestionModel.
+
+    Returns:
+        Suggestion. The corresponding Suggestion domain object.
+    """
+    return suggestion_domain.Suggestion(
+        suggestion_model.id, suggestion_model.suggestion_type,
+        suggestion_model.target_type, suggestion_model.target_id,
+        suggestion_model.target_version_at_submission,
+        suggestion_model.status, suggestion_model.author_id,
+        suggestion_model.assigned_reviewer_id,
+        suggestion_model.final_reviewer_id, suggestion_model.change_cmd,
+        suggestion_model.score_category)
+
+
+def get_suggestion_by_id(suggestion_id):
+    """Finds a suggestion by the suggestion ID.
+
+    Args:
+        suggestion_id: str. The ID of the suggestion.
+
+    Returns:
+        Suggestion, or None if no suggestion is found.
+    """
+    model = suggestion_models.SuggestionModel.get_by_id(suggestion_id)
+
+    return get_suggestion_from_model(model) if model else None
+
+
+def get_suggestions_by_author(author_id):
+    """Gets a list of suggestions by the given author.
+
+    Args:
+        author_id: str. The author of the suggestions.
```

Comment:
Here and below: in docstring descriptions, differentiate clearly between the user ID and the user themselves.

Reply:
Done